purchase_date_planned_manual_container_deposit_product/models/purchase_order.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def prepare_deposit_container_line(self, product, qty):
+        values = super().prepare_deposit_container_line(product, qty)
+        if self.date_planned:
```

Comment:
A comment is welcome, why the `values["date_planned"]` is overridden here and we need to fix it?

Reply:
`prepare_deposit_container_line` will return values to create a new purchase order line with the container deposit product. If we enter Expected Arrival `date_planned` in the purchase order, we need to send the date_planned from the purchase order to the purchase order line. If not, the new purchase order line will have date_planned as today.

The  [method](https://github.com/OCA/product-attribute/blob/16.0/product_packaging_container_deposit/models/container_deposit_order_mixin.py#L18) I override and [where](https://github.com/OCA/product-attribute/blob/16.0/product_packaging_container_deposit/models/container_deposit_order_mixin.py#L89) it was executed.